tools/ci_build/github/azure-pipelines/mac-coreml-ci-pipeline.yml

```
@@ -1,21 +1,34 @@
 jobs:
 - job: CoreML_CI
+  workspace:
+    clean: all
   pool:
     vmImage: 'macOS-12'
   variables:
     MACOSX_DEPLOYMENT_TARGET: '10.14'
+    TODAY: $[format('{0:dd}{0:MM}{0:yyyy}', pipeline.startTime)]
+    CCACHE_DIR: '$(Pipeline.Workspace)/ccache'
   timeoutInMinutes: 120
   steps:
   - script: brew install coreutils ninja
     displayName: Install coreutils and ninja
 
-  - script: |
-      python3 tools/ci_build/build.py \
-      --build_dir build \
-      --skip_submodule_sync \
-      --cmake_generator=Ninja \
-      --parallel \
-      --build_shared_lib \
-      --config Debug \
-      --use_coreml
-    displayName: CoreML EP, Build and Test on macOS
\ No newline at end of file
+  - template: templates/mac-build-step-with-cache.yml
+    parameters:
+      TODAY: $(TODAY)
+      AdditionalKey: coreml
+      CACHE_DIR: $(CCACHE_DIR)
+      BuildStep:
+        - script: |
+            python3 tools/ci_build/build.py \
+            --build_dir build \
+            --skip_submodule_sync \
+            --cmake_generator=Ninja \
+            --parallel \
+            --build_shared_lib \
+            --config Debug \
+            --use_cache \
+            --use_coreml
+          displayName: CoreML EP, Build and Test on macOS
+          env:
+            CCACHE_COMPILERCHECK: content
```

nit: extra indentation
```suggestion
      - script: |
          python3 tools/ci_build/build.py \
          --build_dir build \
          --skip_submodule_sync \
          --cmake_generator=Ninja \
          --parallel \
          --build_shared_lib \
          --config Debug \
          --use_cache \
          --use_coreml
        displayName: CoreML EP, Build and Test on macOS
        env:
          CCACHE_COMPILERCHECK: content
```